ext/pgsql/tests/pg_result_memory_size.phpt

```diff
@@ -0,0 +1,29 @@
+--TEST--
+pg_result_memory_size
+--EXTENSIONS--
+pgsql
+--SKIPIF--
+<?php
+include("inc/skipif.inc");
+if (!function_exists('pg_result_memory_size')) die('skip function pg_result_memory_size() does not exist');
+?>
+--FILE--
+<?php
+include('inc/config.inc');
+
+$db = pg_connect($conn_str);
+
+$result = pg_query($db, 'select 1');
```

Comment:
question: is the memory usage deterministic right ? If that s the case if I redo this query it should be the same amount correct ?

Reply:
The memory usage returned by `PQresultMemorySize()` for the same query is generally consistent, but this cannot be guaranteed in all cases due to several factors:

* **Version of libpq:** Although significant differences are unlikely, the lack of documented specifications of `libpq` means we cannot ensure identical behavior across all versions.
* **CPU architecture:** Differences in architecture could potentially influence memory usage.
* **Query purity:** Non-deterministic elements, such as the output of `VOLATILE` functions or results from `INSERT INTO .. RETURNING ...` / `UPDATE ... RETURNING`, can affect the memory used.

Based on some tests with simple queries, the memory usage has been consistent when none of these variables are at play. However, I must note that I am not deeply familiar with the internal implementation of `libpq`, so while my response is based on research and tests, it might not cover all possible scenarios.

(For the above reasons, I did not write the output size directly in the test, but instead tested by comparing the two results)

Reply:
Thanks for the detail but what I meant if within your test if I do

```php
$db = pg_connect($conn_str);
$result = pg_query($db, 'select 1');
$size_3 = pg_result_memory_size($result);
var_dump($size_1 == $size_3);
```

it should be true then ?

Reply:
Yes, `true` in that case.
(Although not guaranteed, I don't think this behavior will change.)